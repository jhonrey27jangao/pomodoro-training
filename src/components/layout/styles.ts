import styled from "styled-components";

export const StyledLayoutContainer = styled.div`
  background-color: #f2f5fa;
  background-image: url(https://pixelarity.com/items/demos/vortex/magenta/images/overlay.png),
    url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0naHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmcnIHdpZHRoPScyMDAnIGhlaWdodD0nMTAwJyB2aWV3Qm94PScwIDAgMjAwIDEwMCcgcHJlc2VydmVBc3BlY3RSYXRpbz0nbm9uZSc+PGRlZnM+PHJhZGlhbEdyYWRpZW50IGlkPSdyZzEnIGN4PScwJScgY3k9JzAlJyByPScxMDAlJyBmeD0nMCUnIGZ5PScwJSc+PHN0b3Agb2Zmc2V0PScwJScgc3R5bGU9J3N0b3AtY29sb3I6ICNhNDgwZmY7IHN0b3Atb3BhY2l0eTogMC41JyAvPjxzdG9wIG9mZnNldD0nMTAwJScgc3R5bGU9J3N0b3AtY29sb3I6ICNhNDgwZmY7IHN0b3Atb3BhY2l0eTowJyAvPjwvcmFkaWFsR3JhZGllbnQ+PHJhZGlhbEdyYWRpZW50IGlkPSdyZzInIGN4PScxMDAlJyBjeT0nMCUnIHI9JzEwMCUnIGZ4PScxMDAlJyBmeT0nMCUnPjxzdG9wIG9mZnNldD0nMCUnIHN0eWxlPSdzdG9wLWNvbG9yOiAjZmY4ZjgzOyBzdG9wLW9wYWNpdHk6IDAuNScgLz48c3RvcCBvZmZzZXQ9JzEwMCUnIHN0eWxlPSdzdG9wLWNvbG9yOiAjZmY4ZjgzOyBzdG9wLW9wYWNpdHk6MCcgLz48L3JhZGlhbEdyYWRpZW50PjwvZGVmcz48cG9seWdvbiBwb2ludHM9JzAsMCAxMjUsMCAxMjUsMTAwIDAsMTAwJyBzdHlsZT0nZmlsbDogdXJsKCNyZzEpOycgLz48cG9seWdvbiBwb2ludHM9Jzc1LDAgMjAwLDAgMjAwLDEwMCA3NSwxMDAnIHN0eWxlPSdmaWxsOiB1cmwoI3JnMik7JyAvPjwvc3ZnPg==),
    url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0naHR0cDovL3d3dy53My5vcmcvMjAwMC9zdâ€¦AyMDAsMCAyMDAsMTAwIDc1LDEwMCcgc3R5bGU9J2ZpbGw6IHVybCgjcmcyKTsnIC8+PC9zdmc+);
  background-size: auto, 100% 60rem, 100% 60rem;
  background-position: top left, top left, bottom left;
  background-repeat: repeat, no-repeat, no-repeat;
  font-family: "Source Sans Pro", Helvetica, sans-serif !important;
  margin: 0;
  padding: 20px 0 0;
  overflow: hidden;

  & h1,
  h2,
  h3,
  p {
    margin: 0;
    padding: 0;
  }

  & div,
  section {
    box-sizing: border-box;
  }

  & ul {
    margin: 0;
    padding: 0;
    list-style: none;
    text-align: left;
  }
`;

export const StyledLayoutContent = styled.div`
  width: 80rem;
  max-width: calc(100% - 7rem);
  margin: 0 auto;
  padding: 65px 0;
  overflow: hidden;
`;
