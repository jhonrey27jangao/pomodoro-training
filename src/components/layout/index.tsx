import React from "react";
import { LayoutProps } from "./types";
import { StyledLayoutContainer, StyledLayoutContent } from "./styles";
import { Navbar } from "../navbar/index";
export const Layout: React.FC<LayoutProps> = ({ children }) => (
  <>
    <Navbar />
    <StyledLayoutContainer>
      <StyledLayoutContent>{children}</StyledLayoutContent>
    </StyledLayoutContainer>
  </>
);
