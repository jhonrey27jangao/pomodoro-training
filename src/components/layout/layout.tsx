import React from "react";
import { LayoutProps } from "./types";
import { StyledLayoutContainer, StyledLayoutContent } from "./styles";
import { Navbar } from "../navbar/navbar";
export const Layout: React.FC<LayoutProps> = ({ children }) => (
  <>
    <Navbar />
    <StyledLayoutContainer>
      <StyledLayoutContent>{children}</StyledLayoutContent>
    </StyledLayoutContainer>
  </>
);
