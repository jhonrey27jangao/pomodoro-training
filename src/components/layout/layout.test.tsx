import React from 'react'
import {render} from '@testing-library/react';
import {Layout} from './layout'

describe('Test for Layout Component', () => {
    it('should render Layout', () => {
        const {container} = render(
            <Layout/>
        )
        expect(container).toMatchSnapshot();
    })

})