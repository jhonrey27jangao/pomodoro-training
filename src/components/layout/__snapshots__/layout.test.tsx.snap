// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Layout Component should render Layout 1`] = `
<div>
  <header
    class="sc-AxiKw hukFpj"
  >
    <div
      class="sc-AxhCb gjpLLB"
    >
      <a
        href="https://steamuserimages-a.akamaihd.net/ugc/946220022126942778/F7EA1482B7967952540BB8F0CEE0822048BDE466/"
      >
        <strong>
          Dont click here!
        </strong>
         Please?
      </a>
    </div>
    <nav
      class="sc-AxhUy dcVecH"
    >
      <ul>
        <li>
          <a
            href="https://assets.change.org/photos/3/wn/ua/SnwNuAfdCHjPqqi-800x450-noPad.jpg?1554888818"
          >
            Home
          </a>
        </li>
        <li
          class="current"
        >
          <a
            href="https://i.kym-cdn.com/photos/images/newsfeed/001/443/311/70f.gif"
          >
            Pomodoro
          </a>
        </li>
        <li>
          <a
            href="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTTAUTKagxjPG_3C_tbizZOkGZtqiLCYMcVpDTBECW7Ht83jpG_"
          >
            About Mar
          </a>
        </li>
        <li>
          <a
            href="https://img-comment-fun.9cache.com/media/aQMBM88/ajGMNMAl_700w_0.jpg"
          >
            Roxas Shitposting
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    class="sc-AxjAm eGUbaw"
  >
    <div
      class="sc-AxirZ dKRSht"
    />
  </div>
</div>
`;
